### DISCUSSION TITLE
Multicontrol Socket Address Error

### DISCUSSION THREAD

**Discussion by mib7289** [Anders-Holst/xled_plus#32] [Q&A] [answered]
I keep getting the following error trying control multiple strings:

`OSError: [WinError 10048] Only one usage of each socket address (protocol/network address/port) is normally permitted`

The code I am running is:
```
# setup multicontrol
mctr = MultiHighControlInterface([ip1,ip2])

# get the layout and convert to numpy arr [[x0,y0,z0],[x1,y1,z1]...]
layout = mctr.get_led_layout()['coordinates']
layout = convert_layout(layout)

# get number of LEDs
num_leds = len(layout)

# create boolean np arr of LEDs with y loc at or under .5 
t = np.array(layout[:, 1] <= .5)
t = t.reshape((num_leds, 1))

# set some simple colors
red = np.array([0, 255, 0, 0])
green = np.array([0, 0, 255, 0])

# set the colors to np arr, create buffer and send
led_colors = np.where(t, red, green)
buffer = io.BytesIO(np.reshape(led_colors, led_colors.size).tobytes())
mctr.show_rt_frame(buffer)
```

It seems other have been successful with the multicontrol but no one seemed to have this issue.  
I have tried forcing different port numbering but that resulted in one strand not responding.
I appreciate any help, especially if I am just missing something dumb.

**Comment by Anders-Holst** [comment, marked as answer]
This issue has been known for a while: It is the UDP server that hogs the socket address on the local machine, such that you cannot use two UDP servers simultaneously, even if they try to connect to two different led devices. This was corrected in the 'xled' package already in 2021, but there has not been any new release since then, so 'pip install' fetches the old code. If you only use one multicontrol, and no other control object in parallell, I think it should manage anyway. But if you start two different control interfaces (to the same or different leds), then you get this error. 
So the quick fix is to delete all other processes which try to control your leds. (That is, you may need to restart python.) Then try multicontrol again. 
The second fix is to try to download the latest version of 'xled' from github instead of using pip, and see if the problem goes away.
If none of the above helps, I may have a workaround, which employs a really simple UDP server. 
If you download 'xled' from github, and it helps, please let me know, and we can suggest scrool to make a release of the newest version.

**Reply by mib7289** [reply]
Using the latest version of 'xled' from github worked. Thanks!
